File: lib/models/news.dart
class News {
  final int id;
  final String title;
  final String content;
  final String? imageUrl;
  final String? date;

  News({
    required this.id,
    required this.title,
    required this.content,
    this.imageUrl,
    this.date,
  });

  factory News.fromJson(Map<String, dynamic> json) {
    return News(
      id: json['id'] is int ? json['id'] : int.tryParse(json['id'].toString()) ?? 0,
      title: json['title'] ?? '',
      content: json['content'] ?? '',
      imageUrl: json['image_url'],
      date: json['date'] ?? json['created_at'],
    );
  }
}

File: admin_react/src/types/news.ts
export interface NewsItem {
  id: number;
  title: string;
  title_ar: string;
  title_en: string;
  content: string;
  content_ar: string;
  content_en: string;
  image_url: string;
  date?: string;
  created_at?: string;
}

export interface NewsFormData {
  title_ar: string;
  title_en: string;
  content_ar: string;
  content_en: string;
  image_url: string;
}
